spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeletesRewrite.java

```diff
@@ -213,45 +217,61 @@ static class PositionDeletesWriterFactory implements DataWriterFactory {
       this.writeProperties = writeProperties;
     }
 
+    private Table underlyingTable(Table table) {
```

Comment:
I'm a little concerned here because it appears this has two negative impacts:
1. This results in a distributed load of the table metadata
2. This relies on loading from the metadata file location directly, which we're trying to remove from other actions.

Reply:
Yeah I'm a bit confused why we couldn't just leverage the existing table broadcast? If it's a different type, does broadcasting the underlying table work instead?

Reply:
I've updated this so that we fetch the format version not on the executors but rather earlier